File: examples/camera_ir_set_isotherms.cpp
#include "stdio.h"
#include <pthread.h>
#include <cstdlib>

#include"payloadSdkInterface.h"

#if (CONTROL_METHOD == CONTROL_UART)
T_ConnInfo s_conn = {
    CONTROL_UART,
    payload_uart_port,
    payload_uart_baud
};
#else
T_ConnInfo s_conn = {
    CONTROL_UDP,
    udp_ip_target,
    udp_port_target
};
#endif

PayloadSdkInterface* my_payload = nullptr;

void quit_handler(int sig);

int main(int argc, char *argv[]){
	signal(SIGINT,quit_handler);

	// create payloadsdk object
	my_payload = new PayloadSdkInterface(s_conn);

	// init payload
	my_payload->sdkInitConnection();
	printf("Waiting for payload signal! \n");

	my_payload->checkPayloadConnection();

    usleep(1000000);
	// set view source
    printf("Set view source to IR! \n");
	my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_VIEW_SRC, PAYLOAD_CAMERA_VIEW_IR, PARAM_TYPE_UINT32);
    usleep(1000000);

	printf("Enable IR Isotherms with high GAIN, sleep 5s ... \n");
    my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS, PAYLOAD_CAMERA_IR_ISOTHERMS_ENABLE , PARAM_TYPE_UINT32);
    usleep(100000);
    my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS_GAIN, PAYLOAD_CAMERA_IR_ISOTHERMS_HIGH_GAIN , PARAM_TYPE_UINT32);
    usleep(5000000); // sleep 5s

	printf("Switch low GAIN, sleep 5s ... \n");
    my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS_GAIN, PAYLOAD_CAMERA_IR_ISOTHERMS_LOW_GAIN , PARAM_TYPE_UINT32);
    usleep(5000000); // sleep 5s

	printf("Disable IR Isotherms.\n");
    my_payload->setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS, PAYLOAD_CAMERA_IR_ISOTHERMS_DISABLE , PARAM_TYPE_UINT32);
    usleep(100000);

    // close payload interface
    try {
        my_payload->sdkQuit();
    }
    catch (int error){}

    return 0;
}

void quit_handler( int sig ){
    printf("\n");
    printf("TERMINATING AT USER REQUEST \n");
    printf("\n");

    // close payload interface
    try {
        my_payload->sdkQuit();
    }
    catch (int error){}

    // end program here
    exit(0);
}

File: camera_ir_set_isotherms.py
#!/usr/bin/env python
import os
os.environ['MAVLINK20'] = '1'
os.environ['MAVLINK_DIALECT'] = 'ardupilotmega'

import time
import signal
import sys
from pymavlink import mavutil
from libs.payload_sdk import PayloadSdkInterface
from libs.payload_define import *

my_payload = None

# Signal handler for quitting
def quit_handler(sig, frame):
    global my_payload
    print("\nTERMINATING AT USER REQUEST")

    # Close payload interface
    if my_payload:
        try:
            my_payload.sdkQuit()
        except Exception as e:
            print(f"Error while quitting payload: {e}")
    # End program    
    sys.exit(0)

def main():
    global my_payload

    signal.signal(signal.SIGINT, quit_handler)

    # Create payloadsdk object
    my_payload = PayloadSdkInterface()

    # Init payload
    my_payload.sdkInitConnection()
    print("Waiting for payload signal!\n")

    # Check connection
    my_payload.checkPayloadConnection()

    # Set view source to IR
    print("Set view source to IR!")
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_VIEW_SRC, payload_camera_view_src.PAYLOAD_CAMERA_VIEW_IR, mavutil.mavlink.MAV_PARAM_TYPE_UINT32)
    time.sleep(1) 

    # Enable IR isotherms with high gain
    print("Enable IR Isotherms with high GAIN, sleep 5s ...")
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS, payload_camera_ir_isotherms.PAYLOAD_CAMERA_IR_ISOTHERMS_ENABLE, mavutil.mavlink.MAV_PARAM_TYPE_UINT32)
    time.sleep(0.1) 
    
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS_GAIN, payload_camera_ir_isotherms_gain.PAYLOAD_CAMERA_IR_ISOTHERMS_HIGH_GAIN, mavutil.mavlink.MAV_PARAM_TYPE_UINT32)
    time.sleep(5)  

    # Switch to low gain
    print("Switch low GAIN, sleep 5s ...")
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS_GAIN, payload_camera_ir_isotherms_gain.PAYLOAD_CAMERA_IR_ISOTHERMS_LOW_GAIN, mavutil.mavlink.MAV_PARAM_TYPE_UINT32)
    time.sleep(5) 

    # Disable IR Isotherms
    print("Disable IR Isotherms.")
    my_payload.setPayloadCameraParam(PAYLOAD_CAMERA_IR_ISOTHERMS, payload_camera_ir_isotherms.PAYLOAD_CAMERA_IR_ISOTHERMS_DISABLE, mavutil.mavlink.MAV_PARAM_TYPE_UINT32)
    time.sleep(0.1) 

    # Close payload interface
    try:
        my_payload.sdkQuit()
        print("Payload connection closed successfully.")
    except Exception as e:
        print(f"Error while quitting payload: {e}") 

if __name__ == "__main__":
    main()